staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -41,6 +41,18 @@ var (
 		[]string{"verb", "host"},
 	)
 
+	// resolverLatency is a Prometheus Histogram metric type partitioned by
+	// "network", and "address" labels. It is used for the rest client DNS resolver latency metrics.
+	resolverLatency = k8smetrics.NewHistogramVec(
+		&k8smetrics.HistogramOpts{
+			Name:           "rest_client_resolver_duration_seconds",
+			Help:           "DNS resolver latency in seconds. Broken down by network, and address.",
+			StabilityLevel: k8smetrics.ALPHA,
+			Buckets:        []float64{0.005, 0.025, 0.1, 0.25, 0.5, 1.0, 2.0, 4.0, 8.0, 15.0, 30.0, 60.0},
```

Comment:
60 seconds sounds very big for the dialer timeout. Looking into the codebase, the only place I am seeing where we are setting the dialer timeout is from the transport cache and the default is 30s: https://github.com/kubernetes/kubernetes/blob/5b92e46b2238b4d84358451013e634361084ff7d/staging/src/k8s.io/client-go/transport/cache.go#L90-L96

@aojea do you perhaps have better boundaries in mind?

Reply:
I have doubts we need these metrics, per your problem statement

> nd DNS resolving takes 4s, then the request might timeout, but the reason why it failed is hidden from admins.

the reasons is not hidden, DNS errors AFAIK are surfaced as that by golang 

> ead udp 172.17.21.51:34254->172.17.21.1:53: i/o timeout

Reply:
Yeah the DNS errors are surfaced by go, but the only way for cluster admins to find out that there is an issue  with the DNS server would be to look at the clients logs. They lack a top-level information that they can look at and take action on.

It is for sure better to look at the DNS server metrics, but the whole point of having metrics in the client is to treat network as a black box, so we can't just assume that the DNS server will be monitored. The more we can surface from the client, the more debugging/reporting options admins will have.

Reply:
fair enough